From: Matthias Brugger <mbrugger@suse.com>
Subject: kABI: protect struct ethtool_link_settings
Patch-mainline: never, kabi
References: bsc#1085050 

Commit 19cab8872692 ("net: ethtool: Add back transceiver type")
adds the transceiver field to the struct ethtool_link_settings.
Ignore this change for KABI, as it does not change the size of the struct.

Signed-off-by: Matthias Brugger <mbrugger@suse.com>
---
 include/uapi/linux/ethtool.h |    4 ++++
 1 file changed, 4 insertions(+)

--- a/include/uapi/linux/ethtool.h
+++ b/include/uapi/linux/ethtool.h
@@ -1732,9 +1732,13 @@ struct ethtool_link_settings {
 	__u8	eth_tp_mdix;
 	__u8	eth_tp_mdix_ctrl;
 	__s8	link_mode_masks_nwords;
+#ifndef __GENKSYMS__
 	__u8	transceiver;
 	__u8	reserved1[3];
 	__u32	reserved[7];
+#else
+	__u32   reserved[8];
+#endif
 	__u32	link_mode_masks[0];
 	/* layout of link_mode_masks fields:
 	 * __u32 map_supported[link_mode_masks_nwords];
